api/src/main/java/io/opencensus/stats/AggregationData.java

```diff
@@ -490,13 +507,37 @@ public double getMax() {
        * @param attachments the contextual information about the example value.
        * @return an {@code Exemplar}.
        * @since 0.16
+       * @deprecated in favor of {@link #createNew(double, Timestamp, Map)}.
        */
+      @Deprecated
       public static Exemplar create(
           double value, Timestamp timestamp, Map<String, String> attachments) {
         Utils.checkNotNull(attachments, "attachments");
-        Map<String, String> attachmentsCopy =
-            Collections.unmodifiableMap(new HashMap<String, String>(attachments));
-        for (Entry<String, String> entry : attachmentsCopy.entrySet()) {
+        Map<String, AttachmentValue> attachmentsCopy = new HashMap<String, AttachmentValue>();
+        for (Entry<String, String> entry : attachments.entrySet()) {
+          Utils.checkNotNull(entry.getKey(), "key of attachments");
+          String string = Utils.checkNotNull(entry.getValue(), "value of attachments");
+          attachmentsCopy.put(entry.getKey(), AttachmentValueString.create(string));
+        }
+        return new AutoValue_AggregationData_DistributionData_Exemplar(
+            value, timestamp, Collections.unmodifiableMap(attachmentsCopy));
+      }
+
+      /**
+       * Creates an {@link Exemplar}.
+       *
+       * @param value value of the {@link Exemplar} point.
+       * @param timestamp the time that this {@code Exemplar}'s value was recorded.
+       * @param attachments the contextual information about the example value.
+       * @return an {@code Exemplar}.
+       * @since 0.20
+       */
+      public static Exemplar createNew(
```

Comment:
Because we already have an API with the same signature: https://github.com/census-instrumentation/opencensus-java/blob/ffd217631bde90866e98ceb5cd98209f72332642/api/src/main/java/io/opencensus/stats/AggregationData.java#L494-L495
The only difference is the type argument in `Map` and Java doesn't allow us to overload a method with just that.

Another option is to just remove the old API. This will be a breaking change but I don't think anyone except the Stats exporters are depending on this.

Reply:
Same argument for getAttachments, what do you think? I think removing this is the best option, we don't break any instrumented code with this.

Reply:
Agree that removing the deprecated APIs are the easiest. I won't expect any client code depending on those APIs since no backend supported Exemplar before. Going to make the change.